pkg/autodetect/main.go

```diff
@@ -147,6 +154,41 @@ func (b *Background) detectClusterRoles() {
 	}
 }
 
+func (b *Background) cleanDeployments() error {
+	log.Info("Detecting orphaned deployments.")
+	deployments := &appsv1.DeploymentList{}
+	opts := &client.ListOptions{}
+	instances := &v1.JaegerList{}
+
+	// get all deployments
+	if err := b.cl.List(context.Background(), opts, deployments); err != nil {
+		return err
```

Comment:
Related to the previous comment: log the message here, and `return` (without an error).